### PR TITLE
compatible with OSX

### PR DESCRIPTION
**Pull request by myvyang**
OSX does not support /proc/cpuinfo

### CODE REVIEW

File: scripts/plot.py [hunk 1]
```diff
@@ -4,13 +4,17 @@
...
+import sys
 import argparse
 import numpy as np
 import matplotlib.pyplot as plt
 import matplotlib.patches as patches
 from matplotlib.backends.backend_pdf import PdfPages
 
 def get_cpu_name():
+    if 'darwin' == sys.platform:
+        return os.popen("sysctl -n machdep.cpu.brand_string").read().strip()
```
Comment (StephanvanSchaik) [inline scripts/plot.py]: Could you please add this in a single commit: "darwin: script: implement get_cpu_name() using sysctl"?

File: source/args.c [hunk 2]
```diff
@@ -233,7 +233,7 @@ int parse_args(struct args *args, int argc, const char *argv[])
 		{ "runs", required_argument, 0, OPTION_RUNS },
 		{ "threshold", required_argument, 0, OPTION_THRESHOLD },
 		{ "output", required_argument, 0, OPTION_OUTPUT },
-		{ 0 },
+		{ NULL, 0, 0, 0 },
```
Comment (StephanvanSchaik) [inline source/args.c:236]: Which compiler are you using and does this result in a warning/error? If this is problematic, then more files have to be changed, and this should end up being a single commit changing these files: "add proper NULL-termination to arrays".

File: source/revanc.c [hunk 3]
```diff
@@ -101,7 +101,7 @@ int brute_force_evict_set(struct page_format *fmt, void *evict_target,
 				slot &= level->slot_mask;
 				expected_slot &= level->slot_mask;
 
-				if (fabs(slot - expected_slot) <= 1.0) {
```
Comment (StephanvanSchaik) [inline source/revanc.c]: I recall casting these values to a float in an earlier version when I was testing this, but I might have accidentally omitted the type cast. Removing the `fabs()` is definitely not correct.

File: source/args.c [hunk 4]
```diff
@@ -247,7 +247,7 @@ int parse_args(struct args *args, int argc, const char *argv[])
 		case OPTION_TARGET:
 			if ((parse_addr(&args->target, optarg)) < 0)
 				return -1;
-			
+
```
Comment (StephanvanSchaik) [inline source/args.c:250]: This is a formatting change. If there are many of those, it makes more sense to have this as a separate commit: "formatting: remove whitespace from empty lines".

**Review by StephanvanSchaik — CHANGES_REQUESTED** [review]

File: scripts/plot.py (hunk 1 above)
Comment (myvyang) [reply scripts/plot.py]: see https://github.com/vusec/revanc/pull/6

File: source/args.c (hunk 2 above)
Comment (myvyang) [reply source/args.c:236]: gcc and clang on OSX report a warning(-Wmissing-field-initializers) for this.

```
source/args.c:236:7: warning: missing field 'has_arg' initializer [-Wmissing-field-initializers]
                { 0 },
                    ^
source/x86-64/paging.c:44:9: warning: missing field 'levels' initializer [-Wmissing-field-initializers]
        { NULL },
               ^
```

Comment (StephanvanSchaik) [reply source/args.c:236]: Ah, all right :). Could you also change this for source/arm/paging.c and source/arm64/paging.c? The changes should be pretty much identical for those files as well.

Comment (myvyang) [reply source/args.c:236]: updated.

File: source/args.c (hunk 4 above)
Comment (myvyang) [reply source/args.c:250]: my editor replaces this automatically.  just ignore it.

**Pull request merged**